src/sentry/static/sentry/app/utils/discover/discoverQuery.tsx

```diff
@@ -87,18 +100,42 @@ class DiscoverQuery extends React.Component<Props, State> {
     );
   };
 
-  fetchData = () => {
-    const {eventView, orgSlug, location, limit, keyTransactions} = this.props;
+  getRoute(keyTransactions, trendsType) {
+    if (keyTransactions) {
+      return 'key-transactions';
+    }
+    if (trendsType) {
+      return 'events-trends';
+    }
+    return 'eventsv2';
```

Comment:
At what point should we consider making different components that do different kinds of queries?

Reply:
I've thought about it before putting this up, ultimately it's light branching allowing trends to keep aligned with discover. I think if we were to start to meddle with existing params it'd be a good case to break it out. Aside from a small handful of additional params I don't see trends becoming significantly more complex so I've opted to leave it for the time being.